confdgnmi/testtool/README.md

```diff
@@ -0,0 +1,122 @@
+
+as a parameter in command below to run the test suites:
+
+```bash
+robot --variablefile MY.yaml ./
+```
+
+You may want to run only subset of tests. See `robot`'s `--help` output for details on customization the test-run.
+
+We recommend primarily the `--include` and/or `--exclude` options to allow selecting specific test-cases depending on existing test "**tags**". See the `report.html`/`log.html` or actual `.robot` files  for list of existing tags.
+
+"Successfully" executed (irrespective of actual test results) test run results in output into console like:
+
+```text
+==============================================================================
+Testtool :: Telemetry testing suite for a brief verification of the device'...
+==============================================================================
+Testtool.gNMI Interface :: Test suite dedicated to verification of generic ...
+==============================================================================
+Testtool.gNMI Interface.Capabilities :: Generic device agnostic test suite ...
+==============================================================================
+Device sends CapabilityResponse (any) :: Device can respond with `... | PASS |
+------------------------------------------------------------------------------
+Mandatory JSON is advertised as supported encoding :: Get the list... | FAIL |
+[ JSON_IETF | ASCII | PROTO ] does not contain value 'JSON'.
+------------------------------------------------------------------------------
+[ WARN ] Mandatory JSON (as per gNMI specification) not declared as supported
+Supported encodings include JSON or JSON_IETF :: Test if the devic... | PASS |
+------------------------------------------------------------------------------
+Device should support some models :: Check that the ``supported_mo... | PASS |
+------------------------------------------------------------------------------
+
+...output continued...
+
+Output:  .../output.xml
+Log:     .../log.html
+Report:  .../report.html
+```
+
+This results in creation of previously mentioned log files of interest - `report.html` and `log.html`.
+
+### running without target device - ConfD based "adapter"
+
+You can run tests without having any target device, e.g. to see an example test outputs/logs. You can do this by using the so-called "**ConfD gNMI adapter**" as a target device.
+
+ConfD gNMI adapter serves as an proof-of-concept implementation of gNMI enabled device. It utilizes the ConfD to run and service the model-specific requests for some of OpenConfig standard models<span style="background-color:red">, and TODO...?</span> It includes a simple gNMI server written in Python to "proxy' the incoming gNMI requests into the ConfD database.
```

Comment:
And DEMO version for general gNMI tests.

Reply:
added info about demo mode...